### PR TITLE
Add GNU gettext

### PR DESCRIPTION
**Pull request by codingjoe**
[gettext][gettext] is commonly used in various frameworks and
programming languages to provide support for i18n. Such as
* Python/Django
* Ruby/translation
* Python/flask-babel
* ... may more.

I does not make sense to have it available in side the runtime dyno, but during build time, it is helpful to have the package available to compile gettext `po` files (to `mo`).

There are already some buildpacks doing that, but since this is a GNU tool, it makes sense to have it in the main image.

[gettext]: https://www.gnu.org/software/gettext/

### CODE REVIEW

**Comment by codingjoe** [comment]
@tt what needs to be done to get this thing on the road? Anything I can do?

File: heroku-16-build/bin/heroku-16-build.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@ apt-get install -y --force-yes \
     autoconf \
     bison \
     build-essential \
+    gettext \
```
Comment (ojacobson) [inline heroku-16-build/bin/heroku-16-build.sh]: Hi, and thanks for the contribution!

One question. Putting the actual `gettext` package here will install the `.so` files in the `heroku-16-build` image, but not in the `heroku-16` image. The library will be available at build-time only. Is this intended?

Comment (codingjoe) [reply heroku-16-build/bin/heroku-16-build.sh]: Yes, one should compile gettext files only during deploy not during runtime.

**Comment by dmathieu** [comment]
@codingjoe could you give us the size of the package when installed?

**Comment by codingjoe** [comment]
> @codingjoe could you give us the size of the package when installed?
@dmathieu what's the recommended why to measure that? Only package or including dependencies?

**Comment by edmorley** [comment]
The Travis script outputs the image sizes to the log to help with decisions like these :-)

master: 
       heroku/heroku:16         461MB
       heroku/heroku:16-build   852MB

HEAD~2 (this PR's parent until it's rebased, and so doesn't contain #89):
       heroku/heroku:16         460MB
       heroku/heroku:16-build   851MB

This PR:
       heroku/heroku:16         461MB
       heroku/heroku:16-build   858MB

**Comment by dmathieu** [comment]
Including dependencies. Basically, what's the stack image size difference before and after adding this package.

**Comment by codingjoe** [comment]
6MB than, thanks @ojacobson 
How often/When do you release new images?

**Comment by dmathieu** [comment]
@codingjoe we release stack images when required. It just happens that we're releasing one today (without your patch).
We could release again next week though. I'd like to get some else from @heroku/build to approve the PR though.

**Comment by dmathieu** [comment]
ping @dzuelke too.
WDYT? Especially about having it only in -dev?

**Comment by codingjoe** [comment]
> @codingjoe we release stack images when required. It just happens that we're releasing one today (without your patch).
We could release again next week though. I'd like to get some else from @heroku/build to approve the PR though.

sounds cool, always a pleasure to work with you guys 🥇

**Comment by dzuelke** [comment]
Yeah the dev-only part is fine @dmathieu. Really all that's needed is `msgfmt` but we can only install the entire package. I suppose many people still produce `.mo` files locally and check them into version control, but thinking about it, it actually makes sense to do so at build time.

**Comment by codingjoe** [comment]
> I suppose many people still produce .mo files locally and check them into version control, but thinking about it, it actually makes sense to do so at build time.
My point exactly 👍

**Comment by codingjoe** [comment]
Awesome, docker works. How long does it take until the new image is available during the regular buildpack builds?

**Comment by dmathieu** [comment]
I'll let you know here once I've finished updating the stack image.

**Comment by dmathieu** [comment]
It's out: https://devcenter.heroku.com/changelog-items/1356

**Comment by codingjoe** [comment]
Nice, I do get a `msgfmt: Command not found` but I guess I just need to wait a bit.

**Comment by dmathieu** [comment]
Yes, not all instances have finished updating yet. That'll be done in the next couple of hours.

**Comment by codingjoe** [comment]
I still get `make: msgfmt: Command not found` even after two days. I call `msgfmt` from `post_compile` inside the Python buildpack :/

**Comment by dmathieu** [comment]
@codingjoe we're aware of the issue which is not related to your patch. We're working on it and will let you know once the stack image is shipped again.

**Comment by dmathieu** [comment]
@codingjoe sorry for the delay here. gettext should be available now.

**Comment by codingjoe** [comment]
🎉 I can confirm it works 👍

**Pull request merged**